### PR TITLE
Added `boot.modprobeConfig.useUbuntuModuleBlacklist`.

### PR DESCRIPTION
**Pull request by qhua948**
Allowing disable of the Ubuntu blacklist.

## Description of changes

Added `boot.modprobeConfig.useUbuntuModuleBlacklist`.
Allowing disable of the Ubuntu blacklist.

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Review by tomfitzhenry — APPROVED** [review]
LGTM.

**Review by tomfitzhenry — CHANGES_REQUESTED** [review]
Actually one minor issue.

Per https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#commit-conventions , the commit message should begin with "nixos/modprobe: ".

**Comment by qhua948** [comment]
> Actually one minor issue.
> 
> Per https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#commit-conventions , the commit message should begin with "nixos/modprobe: ".
done

**Review by tomfitzhenry — APPROVED** [review]

**Review by philiptaron — APPROVED** [review]
Heck yes.

**Comment by philiptaron** [comment]
> @ofborg test kernel-generic

sigh.

https://logs.ofborg.org/?key=nixos/nixpkgs.290330&attempt_id=1d0ee1a4-ba23-479c-8ec8-742d61cddc1f

```
       at /ofborg/checkout/repo/38dca4e3aa6bca43ea96d2fcc04e8229/builder/ofborg-evaluator-4/lib/modules.nix:607:52:
          606|                 (n: value:
          607|                   [{ inherit (module) file; inherit value; }]
             |                                                    ^
          608|                 )
       error: linux 4.14 was removed because it will reach its end of life within 23.11
```

... so it didn't run anything.

File: nixos/modules/system/boot/modprobe.nix [hunk 1]
```diff
@@ -11,6 +11,10 @@ with lib;
       default = true;
     };
 
+    boot.modprobeConfig.useUbuntuModuleBlacklist = mkEnableOption (lib.mdDoc "Ubuntu distro's module blacklist") // {
```
Comment (FliegendeWurst) [inline nixos/modules/system/boot/modprobe.nix:14]:
```suggestion
    boot.modprobeConfig.useUbuntuModuleBlacklist = mkEnableOption "Ubuntu distro's module blacklist" // {
```
mdDoc is obsolete.

**Review by FliegendeWurst — APPROVED** [review]

**Pull request merged**